contracts/LockedFund/LockedFund.sol

```diff
@@ -394,72 +503,125 @@ contract LockedFund is ILockedFund {
 	 * @param _duration The duration for vesting.
 	 * @param _basisPoint The % (in Basis Point) which determines how much will be (waited) unlocked immediately.
 	 * @param _unlockedOrWaited Determines if the Basis Point determines the Unlocked or Waited Unlock Balance.
+	 * @param _receiveTokens - True if tokens should be taken from caller, False otherwise.
 	 */
 	function _depositVested(
 		address _userAddress,
 		uint256 _amount,
 		uint256 _cliff,
 		uint256 _duration,
 		uint256 _basisPoint,
-		UnlockType _unlockedOrWaited
+		UnlockType _unlockedOrWaited,
+		bool _receiveTokens
 	) internal {
 		/// If duration is also zero, then it is similar to Unlocked Token.
 		require(_duration != 0, "LockedFund: Duration cannot be zero.");
 		require(_duration <= MAX_DURATION, "LockedFund: Duration is too long.");
 
 		// MAX_BASIS_POINT is not included because if 100% is unlocked, then this function is not required to be used.
 		require(_basisPoint < MAX_BASIS_POINT, "LockedFund: Basis Point has to be less than 10000.");
-		bool txStatus = token.transferFrom(msg.sender, address(this), _amount);
-		require(txStatus, "LockedFund: Token transfer was not successful. Check receiver address.");
-
+		if (_receiveTokens) {
+			bool txStatus = token.transferFrom(msg.sender, address(this), _amount);
+			require(txStatus, "LockedFund: Token transfer was not successful. Check receiver address.");
+		} else {
+			missingBalance = missingBalance.add(_amount);
+		}
 		uint256 unlockedBal = _amount.mul(_basisPoint).div(MAX_BASIS_POINT);
 
 		if (_unlockedOrWaited == UnlockType.Immediate) {
 			unlockedBalances[_userAddress] = unlockedBalances[_userAddress].add(unlockedBal);
-			// TODO emit unlocked deposited event.
+			emit UnlockedDeposited(msg.sender, _userAddress, unlockedBal);
 		} else if (_unlockedOrWaited == UnlockType.Waited) {
 			waitedUnlockedBalances[_userAddress] = waitedUnlockedBalances[_userAddress].add(unlockedBal);
-			// TODO emit waited unlocked deposited event.
+			emit WaitedUnlockedDeposited(msg.sender, _userAddress, unlockedBal);
 		} else {
 			unlockedBal = 0;
 		}
 
-		vestedBalances[_userAddress] = vestedBalances[_userAddress].add(_amount).sub(unlockedBal);
+		/// @notice Creating unique vesting data identifier.
+		bytes32 _vestingData = _getVestingData(_cliff, _duration);
+
+		/// @notice Checking if it already is in the list.
+		bool _vestingExist;
+		for (uint256 i = 0; i < userVestings[_userAddress].length; i++) {
+			if (userVestings[_userAddress][i] == _vestingData) {
+				_vestingExist = true;
+			}
+		}
```

Comment:
Looks very similar to `checkUserVestingsOf` function

Reply:
Solved in 9c7c4c2